xcell/mappers/mapper_DESY3wl.py

```diff
@@ -0,0 +1,446 @@
+        # Weights after the selection cuts have been applied
+        self.weights = dict(zip(self.mcal_groups, nones))
+        # Final ellipticities; i.e. the ones to be used for maps
+        self.ellips_unbiased = {'PSF': None, 'shear': None}
+        self.debug = config.get("debug", False)
+
+    def _check_kind(self, kind):
+        if kind not in self.mcal_groups:
+            raise ValueError(f"kind={kind} not valid. It needs to be one of " +
+                             ", ".join(self.mcal_groups))
+
+    def _get_cat_index(self):
+        # Read the index catalog that links to the other ones.
+        # Columns explained in
+        # https://des.ncsa.illinois.edu/releases/y3a2/Y3key-catalogs
+        if self.cat_index is None:
+            self.cat_index = h5py.File(self.config['indexcat'], mode='r')
+
+        return self.cat_index
+
+    def _get_select(self, kind='unsheared'):
+        self._check_kind(kind)
+        if self.select[kind] is None:
+            index = self._get_cat_index()
+            subgroup = "select"
+            subgroup += kind[-3:] if kind != 'unsheared' else ''
+            # select is an array of indices
+            select = index[f'index/{subgroup}_bin{self.zbin+1}'][:]
+            if self.debug:
+                select = select[:10000]
+            self.select[kind] = select
+
+        return self.select[kind]
+
+    def _get_ellips(self, kind='unsheared', mode=None):
+        self._check_kind(kind)
+        _, _, mode = self._set_mode(mode)
+        if self.ellips[mode][kind] is None:
+            e1f, e2f, mode = self._set_mode(mode)
+            sel = self._get_select(kind)
+            index = self._get_cat_index()
+            # For some reason [:][sel] is faster than [sel]
+            e1 = index[f'catalog/metacal/{kind}'][e1f][:][sel]
+            e2 = index[f'catalog/metacal/{kind}'][e2f][:][sel]
+            self.ellips[mode][kind] = np.array((e1, e2))
+
+        return self.ellips[mode][kind]
+
+    def get_ellips_unbiased(self, mode=None):
+        _, _, mode = self._set_mode(mode)
+        if self.ellips_unbiased[mode] is None:
+            ellips = self._get_ellips(mode=mode).copy()
+            # TODO: Only for shear??
+            if mode == 'shear':
+                # Following 2203.07128
+                # Note that, prior to eq. (1), observed ellipticities were
+                # corrected for additive and multiplicative biases by
+                # subtracting the (weighted) mean ellipticity (as done in
+                # Gatti, Shel- don et al. 2021c) and dividing by the
+                # Metacalibration response, both of which were computed for
+                # each bin
+
+                # Remove additive bias
+                # ellips -= np.mean(ellips, axis=1)[:, None]
+                w = self.get_weights()
+                c = np.average(ellips, weights=w, axis=1)[:, None]
+                print(f"Additive bias: {c}")
```

Comment:
Do we need this?

Reply:
Do you mean the print statement? I find it handy to check against the values in the official paper

Reply:
ok, but you have a similar thing for the multiplicative one that you commented out.